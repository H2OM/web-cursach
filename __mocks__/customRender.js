import ClientContext from '@/lib/context/ClientContext';
import { render } from '@testing-library/react';

export default customRender = (ui, {providerProps}) => {
    return render(
      <ClientContext.Provider value={{...providerProps}}>{ui}</ClientContext.Provider>,
    )
}