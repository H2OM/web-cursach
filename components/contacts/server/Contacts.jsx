import Uptitle from "@/lib/basecomponents/uptitle/uptitle";
import Image from 'next/image';
import '../contacts.scss';
import Miniform from "@/lib/basecomponents/miniform/miniform";

export default function Contacts () {

    return (
        <section className="Contacts">
            <div className="container">
                <Uptitle full><h1 className="title">Контакты</h1></Uptitle>
                <div className="Contacts__grid">
                    <div className="Contacts__grid__section">
                        <h2 className="subtitle">TopAttractions@gmail.com</h2>
                        <h2 className="subtitle">+7 900 111-11-11</h2>
                        <div className="Contacts__grid__section__icons">
                            <Image
                                src="/svg/icon_telegram.svg"
                                alt="tg"
                                className="Contacts__grid__section__icons__icon"
                                width={0}
                                height={0}
                                sizes='100vw'
                                priority={true}
                                quality={100}
                            />
                            <Image
                                src="/svg/icon_vk.svg"
                                alt="vk"
                                className="Contacts__grid__section__icons__icon"
                                width={0}
                                height={0}
                                sizes='100vw'
                                priority={true}
                                quality={100}
                            />
                        </div>
                    </div>
                    <div className="Contacts__grid__section">
                        <h2 className="subtitle">Подпишитесь на рассылку новостей</h2>
                        <Miniform size/>
                    </div>
                </div>
            </div>
        </section>
    )
}