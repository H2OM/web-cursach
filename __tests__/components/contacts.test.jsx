import Contacts from '@/components/contacts/server/Contacts';
import 'intersection-observer';
import customRender from '@/__mocks__/customRender';



it('contacts snapshot', async () => {
  const {container} = customRender(<Contacts/>, {});
  expect(container).toMatchSnapshot();
});