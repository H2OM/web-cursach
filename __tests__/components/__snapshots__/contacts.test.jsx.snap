// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contacts snapshot 1`] = `
<div>
  <section
    class="Contacts"
  >
    <div
      class="container"
    >
      <div
        class="Uptitle__wrap"
      >
        <div
          class="Uptitle__anchorUpUptitle__anchorUp_full "
        />
        <div
          class="Uptitle Uptitle_full "
        >
          <h1
            class="title"
          >
            Контакты
          </h1>
        </div>
        <div
          class="Uptitle__anchorDownUptitle__anchorDown_full "
        />
      </div>
      <div
        class="Contacts__grid"
      >
        <div
          class="Contacts__grid__section"
        >
          <h2
            class="subtitle"
          >
            TopAttractions@gmail.com
          </h2>
          <h2
            class="subtitle"
          >
            +7 900 111-11-11
          </h2>
          <div
            class="Contacts__grid__section__icons"
          >
            <img
              alt="tg"
              class="Contacts__grid__section__icons__icon"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="0"
              src="/svg/icon_telegram.svg"
              style="color: transparent;"
              width="0"
            />
            <img
              alt="vk"
              class="Contacts__grid__section__icons__icon"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="0"
              src="/svg/icon_vk.svg"
              style="color: transparent;"
              width="0"
            />
          </div>
        </div>
        <div
          class="Contacts__grid__section"
        >
          <h2
            class="subtitle"
          >
            Подпишитесь на рассылку новостей
          </h2>
          <form
            class="Miniform Miniform_size "
          >
            <div
              class="Miniform__input__wrap Miniform__input__wrap_size "
            >
              <input
                class="Miniform__input "
                name="mail"
                placeholder="Ваша почта"
                required=""
                type="text"
              />
              <label
                class="Miniform__input__error "
                for="email"
              >
                Неверный формат почты
              </label>
            </div>
            <button
              class="Miniform__button "
              type="submit"
            >
              Подписаться
            </button>
          </form>
        </div>
      </div>
    </div>
  </section>
</div>
`;
